src/coreclr/jit/utils.h

```diff
@@ -160,6 +160,18 @@ int signum(T val)
     }
 }
 
+template <typename T, typename U>
+constexpr auto max(T&& t, U&& u) -> decltype(t > u ? t : u)
+{
+    return t > u ? t : u;
+}
+
+template <typename T, typename U>
+constexpr auto min(T&& t, U&& u) -> decltype(t < u ? t : u)
+{
+    return t < u ? t : u;
+}
```

Comment:
Another instance of this pattern. I do not think we should be introducing yet another min/max pattern that is unexpected. What is the cost to properly cast?